marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/OutputEndpointImpl.java

```diff
@@ -79,14 +87,17 @@ public void awaitCompletion() {
                     throw new RuntimeException("error while calling "+getEndpoint().getEndpointPath(), throwable);
                 }
 
+                InputStream[] result = null;
+                if(output != null) {
+                    result = output.toArray(size -> new InputStream[size]);
+                }
                 if (allowsEndpointState()) {
-                    if(output!= null && output.findFirst()!= null && output.findFirst().isPresent()) {
-                        InputStream[] result = output.toArray(size -> new InputStream[size]);
-                        setEndpointState(result[0]);
-                        for(int i=1; i<result.length; i++) {
-                            forEachOutput((Consumer<InputStream>) result[i]);
+                        if(result !=null) {
+                            setEndpointState(result[0]);
                         }
-                    }
+                }
+                for(int i=1; i<result.length; i++) {
```

Comment:
Should i be initialized to 0 if the allowsEndpointState() is false?  In that case, the first result would be output instead of the endpoint state.